diff --git a/src/slic3r/GUI/PartPlate.cpp b/src/slic3r/GUI/PartPlate.cpp
index 3d4333d1..71665f3d 100644
--- a/src/slic3r/GUI/PartPlate.cpp
+++ b/src/slic3r/GUI/PartPlate.cpp
@@ -3724,7 +3724,7 @@ int PartPlateList::create_plate(bool adjust_position)
 	int new_index;
 
 	// make sure belt only have one plater
-	if (wxGetApp().preset_bundle->machine_is_belt())
+	if (wxGetApp().preset_bundle && wxGetApp().preset_bundle->machine_is_belt())
 		if (m_plate_list.size() >= 1)
 			return -1;
 
