diff --git a/src/slic3r/GUI/PartPlate.cpp b/src/slic3r/GUI/PartPlate.cpp
index 3d4333d1..b67b4837 100644
--- a/src/slic3r/GUI/PartPlate.cpp
+++ b/src/slic3r/GUI/PartPlate.cpp
@@ -5044,7 +5044,9 @@ bool PartPlateList::set_shapes(const Pointfs& shape, const Pointfs& exclude_area
 		pos = compute_shape_position(i, m_plate_cols);
 		plate->set_shape(shape, exclude_areas, pos, height_to_lid, height_to_rod);
 	}
-	set_verder_type(wxGetApp().preset_bundle->get_current_vendor_type());
+	if (m_plater && wxGetApp().preset_bundle)
+		set_verder_type(wxGetApp().preset_bundle->get_current_vendor_type());
+
 	calc_bounding_boxes();
 
 	update_logo_texture_filename(texture_filename);
