diff --git a/src/slic3r/CMakeLists.txt b/src/slic3r/CMakeLists.txt
index 753c057..8051839 100644
--- a/src/slic3r/CMakeLists.txt
+++ b/src/slic3r/CMakeLists.txt
@@ -802,5 +802,8 @@ encoding_check(libslic3r_gui)

 target_link_libraries(libslic3r_gui libslic3r cereal::cereal imgui imguizmo minilzo GLEW::GLEW OpenGL::GL hidapi ${wxWidgets_LIBRARIES} glfw  libcurl OpenSSL::SSL OpenSSL::Crypto noise::noise)
+if (NOT WIN32 AND NOT APPLE)
+    target_link_libraries(libslic3r_gui deflate Lerc)
+endif()
 if(WIN32)
     target_link_libraries(libslic3r_gui iphlpapi)
 endif()
