diff --git a/src/slic3r/GUI/GLShadersManager.cpp b/src/slic3r/GUI/GLShadersManager.cpp
index 93083d1caa..ff98e659d6 100644
--- a/src/slic3r/GUI/GLShadersManager.cpp
+++ b/src/slic3r/GUI/GLShadersManager.cpp
@@ -3,5 +3,6 @@
 #include "GLShadersManager.hpp"
 #include "3DScene.hpp"
 #include "GUI_App.hpp"
+#include "OpenGLManager.hpp"
 
 #include <cassert>
@@ -38,7 +38,7 @@ std::pair<bool, std::string> GLShadersManager::init()
     // used to render wireframed triangles
     valid &= append_shader("wireframe", { prefix + "wireframe.vs", prefix + "wireframe.fs" });
 #else
-    const std::string prefix = GUI::wxGetApp().is_gl_version_greater_or_equal_to(3, 1) ? "140/" : "110/";
+    const std::string prefix = GUI::OpenGLManager::get_gl_info().is_version_greater_or_equal_to(3, 1) ? "140/" : "110/";
 #endif // SLIC3R_OPENGL_ES
     // imgui shader
     valid &= append_shader("imgui", { prefix + "imgui.vs", prefix + "imgui.fs" });
@@ -77,3 +77,3 @@ std::pair<bool, std::string> GLShadersManager::init()
-    if (GUI::wxGetApp().is_gl_version_greater_or_equal_to(3, 3)) {
+    if (GUI::OpenGLManager::get_gl_info().is_version_greater_or_equal_to(3, 3)) {
         valid &= append_shader("gouraud_light_instanced", { prefix + "gouraud_light_instanced.vs", prefix + "gouraud_light_instanced.fs" });
     }
diff --git a/src/slic3r/GUI/GLModel.cpp b/src/slic3r/GUI/GLModel.cpp
index f7fab43817..0df2c767fc 100644
--- a/src/slic3r/GUI/GLModel.cpp
+++ b/src/slic3r/GUI/GLModel.cpp
@@ -457,8 +457,10 @@ void GLModel::init_from(const indexed_triangle_set& its)
     data.reserve_indices(3 * its.indices.size());
 
     // Read user preference: smooth normals enabled
-    const bool realistic_mode = wxGetApp().app_config != nullptr && wxGetApp().app_config->get_bool(SETTING_OPENGL_REALISTIC_MODE);
-    const bool smooth_normals_enabled = wxGetApp().app_config != nullptr && wxGetApp().app_config->get_bool(SETTING_OPENGL_PHONG_SMOOTH_NORMALS);
+    const bool realistic_mode = wxTheApp != nullptr && wxGetApp().app_config != nullptr &&
+                                wxGetApp().app_config->get_bool(SETTING_OPENGL_REALISTIC_MODE);
+    const bool smooth_normals_enabled = wxTheApp != nullptr && wxGetApp().app_config != nullptr &&
+                                        wxGetApp().app_config->get_bool(SETTING_OPENGL_PHONG_SMOOTH_NORMALS);
 
     if (realistic_mode && smooth_normals_enabled) {
         // Use per-corner smooth normals (via IGL)
