diff --git a/src/slic3r/GUI/GLShadersManager.cpp b/src/slic3r/GUI/GLShadersManager.cpp
index 229ab3e..3747a78 100644
--- a/src/slic3r/GUI/GLShadersManager.cpp
+++ b/src/slic3r/GUI/GLShadersManager.cpp
@@ -3,5 +3,6 @@
 #include "GLShadersManager.hpp"
 #include "3DScene.hpp"
 #include "GUI_App.hpp"
+#include "OpenGLManager.hpp"
 
 #include <cassert>
@@ -33,7 +33,7 @@ std::pair<bool, std::string> GLShadersManager::init()
 
     bool valid = true;
 
-    bool gl31 = GUI::wxGetApp().is_gl_version_greater_or_equal_to(3, 1);
+    bool gl31 = GUI::OpenGLManager::get_gl_info().is_version_greater_or_equal_to(3, 1);
     const std::string prefix = gl31 ? "140/" : "110/";
 
 	// imgui shader
@@ -57,7 +57,7 @@ std::pair<bool, std::string> GLShadersManager::init()
     // used to render printbed
     valid &= append_shader("printbed", { prefix + "printbed.vs", prefix + "printbed.fs" });
     // used to render options in gcode preview
-    if (GUI::wxGetApp().is_gl_version_greater_or_equal_to(3, 3)) {
+    if (GUI::OpenGLManager::get_gl_info().is_version_greater_or_equal_to(3, 3)) {
         valid &= append_shader("gouraud_light_instanced", { prefix + "gouraud_light_instanced.vs", prefix + "gouraud_light_instanced.fs" });
     }
 
@@ -68,7 +68,7 @@ std::pair<bool, std::string> GLShadersManager::init()
 #endif // ENABLE_ENVIRONMENT_MAP
         );
 
-    if (GUI::wxGetApp().is_gl_version_greater_or_equal_to(3, 3)) {
+    if (GUI::OpenGLManager::get_gl_info().is_version_greater_or_equal_to(3, 3)) {
         // used to render clone shells
         valid &= append_shader("clone_preview_instance", { prefix + "clone_preview_instance.vs", prefix + "clone_preview_instance.fs" });
     }
diff --git a/src/slic3r/GUI/PartPlate.cpp b/src/slic3r/GUI/PartPlate.cpp
index 3d4333d1..57887a63 100644
--- a/src/slic3r/GUI/PartPlate.cpp
+++ b/src/slic3r/GUI/PartPlate.cpp
@@ -4149,7 +4149,8 @@ int PartPlateList::select_plate(int index)
 		//wxQueueEvent(m_plater, new SimpleEvent(EVT_GLCANVAS_PLATE_SELECT));
 	}
 
-    Slic3r::GUI::NotifyAIChatSceneChanged();
+    if (m_plater)
+        Slic3r::GUI::NotifyAIChatSceneChanged();
 	return 0;
 }
 
