diff --git a/build_linux.sh b/build_linux.sh
index 8a5c68f09..a090ea547 100755
--- a/build_linux.sh
+++ b/build_linux.sh
@@ -242,9 +242,9 @@ if [[ -n "${BUILD_ACNEXT}" ]] ; then
     echo "done"
     echo "Building AnycubicSlicerNext ..."
     if [[ -n "${BUILD_DEBUG}" ]] ; then
-        cmake --build build --config Debug --target AnycubicSlicerNext
+        cmake --build build --config Debug --target orca-slicer
     else
-        cmake --build build --config Release --target AnycubicSlicerNext
+        cmake --build build --config Release --target orca-slicer
     fi
     echo "Building AnycubicSlicerNext_profile_validator .."
     if [[ -n "${BUILD_DEBUG}" ]] ; then
