diff --git a/deps/GLFW/GLFW.cmake b/deps/GLFW/GLFW.cmake
index 17a2b8920..c0e6afde4 100644
--- a/deps/GLFW/GLFW.cmake
+++ b/deps/GLFW/GLFW.cmake
@@ -7,8 +7,8 @@ else()
 endif()
 
 if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
-    set(_glfw_use_wayland "-DGLFW_USE_WAYLAND=ON")
+    set(_glfw_use_wayland "-DGLFW_USE_WAYLAND=OFF")
 else()
-    set(_glfw_use_wayland "-DGLFW_USE_WAYLAND=FF")
+    set(_glfw_use_wayland "-DGLFW_USE_WAYLAND=OFF")
 endif()
 
