diff --git a/src/slic3r/GUI/GLShadersManager.cpp b/src/slic3r/GUI/GLShadersManager.cpp
--- a/src/slic3r/GUI/GLShadersManager.cpp
+++ b/src/slic3r/GUI/GLShadersManager.cpp
@@ -3,6 +3,7 @@
 #include "GLShadersManager.hpp"
 #include "3DScene.hpp"
 #include "GUI_App.hpp"
+#include "OpenGLManager.hpp"
 
 #include <cassert>
 #include <algorithm>
@@ -38,7 +39,7 @@ std::pair<bool, std::string> GLShadersManager::init()
     // used to render wireframed triangles
     valid &= append_shader("wireframe", { prefix + "wireframe.vs", prefix + "wireframe.fs" });
 #else
-    const std::string prefix = GUI::wxGetApp().is_gl_version_greater_or_equal_to(3, 1) ? "140/" : "110/";
+    const std::string prefix = GUI::OpenGLManager::get_gl_info().is_version_greater_or_equal_to(3, 1) ? "140/" : "110/";
 #endif // SLIC3R_OPENGL_ES
     // imgui shader
     valid &= append_shader("imgui", { prefix + "imgui.vs", prefix + "imgui.fs" });
@@ -66,7 +67,7 @@ std::pair<bool, std::string> GLShadersManager::init()
     valid &= append_shader("printbed", { prefix + "printbed.vs", prefix + "printbed.fs" });
     valid &= append_shader("hotbed", {prefix + "hotbed.vs", prefix + "hotbed.fs"});
     // used to render options in gcode preview
-    if (GUI::wxGetApp().is_gl_version_greater_or_equal_to(3, 3)) {
+    if (GUI::OpenGLManager::get_gl_info().is_version_greater_or_equal_to(3, 3)) {
         valid &= append_shader("gouraud_light_instanced", { prefix + "gouraud_light_instanced.vs", prefix + "gouraud_light_instanced.fs" });
     }
 
