diff --git a/deps/GLEW/glew/CMakeLists.txt b/deps/GLEW/glew/CMakeLists.txt
index 4cb9de37a..4e138be9c 100644
--- a/deps/GLEW/glew/CMakeLists.txt
+++ b/deps/GLEW/glew/CMakeLists.txt
@@ -4,7 +4,7 @@ project(GLEW)
 
 find_package(OpenGL REQUIRED)
 
-if(OpenGL_EGL_FOUND)
+if(OpenGL_EGL_FOUND AND NOT CMAKE_SYSTEM_NAME STREQUAL "Linux")
     message(STATUS "building GLEW for EGL (hope that wxWidgets agrees, otherwise you won't have any output!)")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DGLEW_EGL")
 endif()
