diff --git a/deps/TIFF/TIFF.cmake b/deps/TIFF/TIFF.cmake
index 0661908f0..83c2cff19 100644
--- a/deps/TIFF/TIFF.cmake
+++ b/deps/TIFF/TIFF.cmake
@@ -17,8 +17,8 @@ if (APPLE)
     )
 else()
     orcaslicer_add_cmake_project(TIFF
-        URL https://gitlab.com/libtiff/libtiff/-/archive/v4.1.0/libtiff-v4.1.0.zip
-        URL_HASH SHA256=c56edfacef0a60c0de3e6489194fcb2f24c03dbb550a8a7de5938642d045bd32
+        URL https://download.osgeo.org/libtiff/tiff-4.1.0.tar.gz
+        URL_HASH SHA256=5d29f32517dadb6dbcd1255ea5bbc93a2b54b94fbf83653b4d65c7d6775b8634
         DEPENDS ${ZLIB_PKG} ${PNG_PKG} dep_JPEG
         CMAKE_ARGS
             -Dlzma:BOOL=OFF
